File: _site/technical/leet/divideArrayIntoEqualPairs.py
# 2206
class Solution:
    def divideArray(self, nums: List[int]) -> bool:
        n = len(nums) // 2

        # gets us the frequency of each number quickly
        freqs = Counter(nums)

        return(all([freq % 2 == 0 for freq in freqs.values()]))

File: docs/technical writing/leet/divideArrayIntoEqualPairs.cpp
//2206
class Solution {
    public:
        bool divideArray(vector<int>& nums) {
            int n = nums.size() / 2;
            unordered_map<int, int> freqs;
            set<int> unique_nums(nums.begin(), nums.end());
    
            for(int num : nums) {
                if(freqs.find(num) != freqs.end() and freqs[num] != 0) {
                    freqs[num]--;
                }
                else {
                    freqs[num]++;
                }
            }
    
            for(auto unum : unique_nums){
                if(freqs[unum] != 0){
                    return(false);
                }
            }
            return(true);
        }
    };